module/behaviour/strategy/PS3Walk/src/PS3Walk.cpp

```diff
@@ -17,123 +17,143 @@
-                                }
-                                break;
-                        }
+                    else {
+                        NUClear::log("Start walking");
+                        moving = true;
                     }
                 }
-                // If it's closed then we should try to reconnect
-                else if (!joystick.valid()) {
-                    joystick.reconnect();
+            });
+
+            on<Trigger<SquareButton>>().then([this](const SquareButton& button) {
+                if (button.pressed) {
+                    if (headLocked) {
+                        NUClear::log("Head unlocked");
+                    }
+                    else {
+                        NUClear::log("Head locked");
+                    }
+                    headLocked = !headLocked;
+                }
+            });
+
+            on<Trigger<CrossButton>>().then([this](const CrossButton& button) {
+                if (button.pressed) {
+                    NUClear::log("Triggering actions");
+                    emit(std::make_unique<MotionCommand>(utility::behaviour::StandStill()));
+                    emit(std::make_unique<ActionPriorites>(ActionPriorites{id, {90}}));
+                    emit(std::make_unique<ExecuteScriptByName>(id, actions));
+                }
+            });
+
+            on<Trigger<CircleButton>>().then([this](const CircleButton& button) {
+                if (button.pressed) {
+                    NUClear::log("Standing");
+                    emit(std::make_unique<MotionCommand>(utility::behaviour::StandStill()));
+                    emit(std::make_unique<ActionPriorites>(ActionPriorites{id, {90}}));
+                    emit(std::make_unique<ExecuteScriptByName>(id, "Stand.yaml"));
+                }
+            });
+
+            on<Trigger<L1Button>>().then([this](const L1Button& button) {
+                if (button.pressed) {
+                    NUClear::log("Requesting Left Front Kick");
+                    emit(std::make_unique<KickScriptCommand>(
+                        KickScriptCommand{Eigen::Vector3d(1, 0, 0),  // vector pointing forward relative to robot
+                                          LimbID::LEFT_LEG}));
+                }
+            });
+
+            on<Trigger<R1Button>>().then([this](const R1Button& button) {
+                if (button.pressed) {
+                    NUClear::log("Requesting Right Front Kick");
+                    emit(std::make_unique<KickScriptCommand>(
+                        KickScriptCommand(Eigen::Vector3d(1, 0, 0),  // vector pointing forward relative to robot
+                                          LimbID::RIGHT_LEG)));
+                }
+            });
+
+            on<Trigger<LeftJoystick>>().then([this](const LeftJoystick& joystick) {
+                switch (joystick.direction.value) {
+                    case LeftJoystick::Direction::HORIZONTAL:
```

Comment:
Do we get separate events for movement in the horizontal and vertical directions?

Reply:
yes